compiler/rustc_passes/src/loops.rs

```diff
@@ -82,6 +99,41 @@ impl<'a, 'hir> Visitor<'hir> for CheckLoopVisitor<'a, 'hir> {
 
     fn visit_expr(&mut self, e: &'hir hir::Expr<'hir>) {
         match e.kind {
+            hir::ExprKind::If(cond, then, else_opt) => {
+                self.visit_expr(cond);
+                if let hir::ExprKind::Block(ref b, None) = then.kind
+                    && matches!(
+                        self.cx_stack.last(),
+                        Some(&Normal)
+                            | Some(&Constant)
+                            | Some(&UnlabeledBlock(_))
+                            | Some(&IfUnlabeledBlock(_))
+                    )
+                {
+                    self.with_context(IfUnlabeledBlock(b.span.shrink_to_lo()), |v| {
+                        v.visit_block(b)
+                    });
+                } else {
+                    self.visit_expr(then);
+                }
+                if let Some(else_expr) = else_opt {
+                    if let hir::ExprKind::Block(ref b, None) = else_expr.kind
```

Comment:
Could this be in a closure used by the if block and the else block?

Reply:
> Could this be in a closure used by the if block and the else block?

Thank you. 

I'm not sure if I understand correctly

 Is this the situation you are referring to?
```rust
fn main() {
        let a = {
        if n == 0 {
            |n| {
                println!("{n}");
                break 1;
            }
        } else {
            |n| {
                println!("{n}");
                break 2;
            }
        }
    };
}
```

For the situation I listed, I think the current diagnostic information is correct:

```bash
error[E0267]: `break` inside of a closure
  --> ./src/main.rs:78:17
   |
76 |             |n| {
   |             --- enclosing closure
77 |                 println!("{n}");
78 |                 break 1;
   |                 ^^^^^^^ cannot `break` inside of a closure

error[E0267]: `break` inside of a closure
  --> ./src/main.rs:83:17
   |
81 |             |n| {
   |             --- enclosing closure
82 |                 println!("{n}");
83 |                 break 2;
   |                 ^^^^^^^ cannot `break` inside of a closure

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0267`.
```

Reply:
> Could this be in a closure used by the if block and the else block?

Sorry. I thought about it again. I misunderstood what you meant before. This section of my code is redundant and repetitive. I will modify it.